### PR TITLE
Enable discarding submodule changes

### PR DESCRIPTION
**Pull request by mislav**
## Overview

Fixes https://github.com/desktop/desktop/issues/4592

## Description

Without `--force`, `git submodule update -- path/to/submodule` is not enough to discard changes in the working copy under the git submodule since git will see that the currently checked out git commit matches the one that the superproject expects and the operation will be a no-op.

Potential TODO:

- [ ] The current approach discards changes under the submodule **without** moving the copies of changed files to Trash. Should we consider moving them to Trash to avoid discarding someone's work irrevocably?
- [ ] It might be a good idea to add a test that exercises discarding changes from multiple submodules at the same time.

## Release notes

Notes:

- Enables discarding changes to files under git submodules

### CODE REVIEW

File: app/src/lib/git/submodule.ts [hunk 1]
```diff
@@ -80,11 +80,9 @@ export async function resetSubmodulePaths(
...
   paths: ReadonlyArray<string>
 ): Promise<void> {
-  for (const path of paths) {
-    await git(
-      ['submodule', 'update', '--recursive', '--', path],
-      repository.path,
-      'updateSubmodule'
-    )
-  }
+  await git(
```
Comment (outofambit) [inline app/src/lib/git/submodule.ts]: doing this in one shell command is super awesome ✨

**Comment by outofambit** [comment]
> It might be a good idea to add a test that exercises discarding changes from multiple submodules at the same time.

:+1: on this if its not too time consuming to add. (i'm not sure if we have some repository with submodules fixtures. @vilmibm and i were looking at unit test setup stuff recently so he can probably answer questions about this too.)

**Comment by outofambit** [comment]
quick note on this one:

> ❌ discarding a Genmoji file when the file name is changed

this is because `git submodule update` doesn't touch untracked files. not sure what we need to do to cover that case too.

that being said, this PR is still a marked improvement on the current state of things, so i'd be in favor of merging as-is and filing a new issue for this last case.

**Comment by niik** [comment]
> * The current approach discards changes under the submodule **without** moving the copies of changed files to Trash. Should we consider moving them to Trash to avoid discarding someone's work irrevocably?

I think we absolutely should consider that but it does seem like this PR improves upon one scenario while not making the trash/restore situation any worse so I still think it's valuable to get this in. I'm gonna open an issue to track this and the fact that discarding a submodule using this method still doesn't clear the dirty state when the submodule contains untracked files.

**Pull request merged**